src/LanguageServer/Microsoft.CodeAnalysis.LanguageServer/HostWorkspace/FileBasedProgramsProjectSystem.cs

```diff
@@ -132,11 +136,20 @@ public async ValueTask TryRemoveMiscellaneousDocumentAsync(DocumentUri uri, bool
 
         var loader = ProjectFactory.CreateFileTextLoader(documentPath);
         var textAndVersion = await loader.LoadTextAsync(new LoadTextOptions(SourceHashAlgorithms.Default), cancellationToken);
-        var (virtualProjectContent, isFileBasedProgram) = VirtualCSharpFileBasedProgramProject.MakeVirtualProjectContent(documentPath, textAndVersion.Text);
+
+        // TODO: consolidate FBP heuristic into new method
+        var (_, isFileBasedProgram) = VirtualProjectXmlProvider.MakeVirtualProjectContent(documentPath, textAndVersion.Text);
+        var content = await _projectXmlProvider.MakeVirtualProjectContentNewAsync(documentPath, cancellationToken);
+        if (content is not var (virtualProjectContent, diagnostics))
+        {
+            return null;
+        }
+
+        // TODO: how to report the diagnostics in the editor?
```

Comment:
@jasonmalinowski I would be interested in any suggestions for how to display the squiggles in the editor for diagnostics returned by run-api, given that SDK is giving us back a full path, display location, and message. We can also punt and open a tracking issue.

Reply:
Punting to #78688